noir-projects/noir-contracts/contracts/test_contract/src/test/note_getter.nr

```diff
@@ -0,0 +1,75 @@
+use crate::test::utils;
+use crate::Test;
+
+global VALUE = 5;
+
+#[test]
+unconstrained fn main() {
+  let (env,
+    test_contract_address,
+    account_1
+  ) = utils::setup();
+  env.impersonate(account_1);
+
+
+  let mut storage_slot = Test::Test_STORAGE_LAYOUT.example_set.slot + 1;
+
+  let mut active_or_nullified = false;
+  env.call_private_void(Test::at(test_contract_address).call_create_note(VALUE, account_1, account_1, storage_slot));
+  env.advance_block_by(1);
+
+  assert_eq(env.call_unconstrained(test_contract_address, || { Test::call_view_notes(storage_slot, active_or_nullified) }), VALUE);
+  assert_eq(env.call_private(Test::at(test_contract_address).call_get_notes(storage_slot, active_or_nullified)), VALUE);
```

Comment:
What's being tested here is a bit unclear. I guess some of PXE is being tested, since it's what TXE uses under the hood. 

At any rate, if splitting this into multiple functions makes the test much slower, this should at the very least have some comments splitting it into sections, as it's quite hard to follow - particularly with the mutable `storage_slot`, which makes one have to guess the intent.